app/scripts/modules/ecs/src/serverGroup/configure/wizard/container/Container.tsx

```diff
@@ -152,13 +153,18 @@ export class Container extends React.Component<IContainerProps, IContainerState>
     this.setState({ targetGroupMappings: currentMappings });
   };
 
+  private updateDirtyTargetGroups = () => {
+    this.props.command.viewState.dirty.targetGroups = null;
+  };
+
   public render(): React.ReactElement<Container> {
     const removeTargetGroupMapping = this.removeTargetGroupMapping;
     const updateContainerMappingImage = this.updateContainerMappingImage;
     const updateTargetGroupMappingTargetGroup = this.updateTargetGroupMappingTargetGroup;
     const updateTargetGroupMappingPort = this.updateTargetGroupMappingPort;
     const updateComputeUnits = this.updateComputeUnits;
     const updateReservedMemory = this.updateReservedMemory;
+    const dirtyTagetGroups = this.props.command.viewState.dirty && this.props.command.viewState.dirty.targetGroups ? this.props.command.viewState.dirty.targetGroups : [];
```

Comment:
Wouldn't dirty.targetGroups be either populated or an empty array? could you do something like this instead:

```
const dirtyTagetGroups = this.props.command.viewState.dirty.targetGroups

Reply:
Sure, I will take care of this.

Reply:
@piradeepk I tried implementing your suggestion but we have kept `targetGroups` as not required (?:) property on `IEcsServerGroupCommandDirty` which makes it to be undefined when it is not yet discovered. If it is undefined then this code breaks and throws **property of undefined** error in later parts of the code. So it is better to have this check at the very start to avoid errors. 
LMK your thought on this.

Reply:
Sounds good! :+1: